homeassistant/components/iotawatt/sensor.py

```diff
@@ -219,6 +220,9 @@ def extra_state_attributes(self) -> dict[str, str]:
         attrs = {"type": data.getType()}
         if attrs["type"] == "Input":
             attrs["channel"] = data.getChannel()
+        if (begin := data.getBegin()) and (last_reset := dt.parse_datetime(begin)):
+            attrs[ATTR_LAST_RESET] = last_reset.isoformat()
```

Comment:
Please set the last reset attribute on the sensor entity instead. It should be a datetime.

https://github.com/home-assistant/core/blob/83fa4df641ab9a0cea82679683905e22494479b1/homeassistant/components/sensor/__init__.py#L256

Note that this feature is only applicable to sensors with state class total.

Reply:
The getter `getBegin()` will only return a value for `"WattHours"` entities. I guess I could double check if its really an entity with state class total in the integration itself instead of relying on `iotawattpy`'s behavior.

Reply:
Addressed with https://github.com/home-assistant/core/pull/65648.

Reply:
Thanks!